### DISCUSSION TITLE
Help! Using Pybricks for Mindstorms NXT & EV3?

### DISCUSSION THREAD

**Discussion by adiggy** [pybricks/support#1383] [Q&A] [answered]
Hi Pybricks crew! 

We bought a Lego Mindstorms NXT 2.0 set from eBay when our son was getting interested in robotics and had just started with the Boost. Much to our dismay, when we went to download the NXT program, we saw that it is no longer compatible with any of our devices (we have Mac laptops 2013-2022 and an Android tablet). 

Furthermore, he’s obsessed with wanting Mindstorms EV3, so we were going to buy a complete eBay set, but we see now that we can’t download that program either. Needless to say we are thrilled we found out about Pybricks and are more than willing to purchase a block coding license, but have some questions before we buy the EV3 set from eBay and commit to being totally dependent on Pybricks block coding.

Questions:
1. On his current Boost app, it has building instructions with hints about how to program in tutorials specific to the models. Will Pybricks  offer the same kind of building and programming instructions for *all* of the advertised models of the NXT and EV3 sets? (We only see building instructions on select models on the website.) 
2. Does Pybricks block coding have the same functionality as the NXT and EV3 software so that if he wants to build models from other builders that use the same sets (like JK Brickworks) he’d be able to follow their programming instructions? 
3. Are all other supporting downloads available through Pybricks that we would need to be able to run the Pybricks block coding program with EV3 and NXT on our devices? (firmware downloads, for example)
4. Anything else we should know about before spending precious resources on retired/unsupported Lego robotics sets for our son? 

Thank you in advance!

**Comment by laurensvalk** [comment, marked as answer]
Pybricks block coding supports Boost, City, Technic, SPIKE Prime, SPIKE Essential, and MINDSTORMS Robot Inventor (51515).

It does not support NXT and EV3 however. For EV3, Pybricks has only Python (not blocks).

If you can find a good deal for MINDSTORMS Robot Inventor (51515), this might also be a good option. You'll be able to use it together with your existing Boost set.

**Reply by adiggy** [reply]
Thank you so much for your quick reply! Super helpful. 🙏
